core/includes/image.inc

```diff
@@ -577,30 +577,39 @@ function image_is_svg($uri) {
  *   which is the accessible mechanism for alternative text within SVGs.
  *
  * @return string
- *   The SVG image contents with the attributes added.
+ *   The SVG image contents with the attributes added. An empty string in case
+ *   of errors.
  *
  * @since 1.28.0 Function added.
  */
 function image_add_svg_attributes($svg_content, array $attributes) {
   $doc = new DOMDocument();
+  libxml_use_internal_errors(TRUE);
   $doc->loadXML($svg_content);
   $svg_tag = $doc->getElementsByTagName('svg')->item(0);
+  $last_error = libxml_get_last_error();
+  libxml_use_internal_errors(FALSE);
+  if (empty($svg_tag)) {
+    $message = $last_error ? trim($last_error->message) : '';
+    watchdog('image', 'Failed to parse SVG content. %message', array(
```

Comment:
It would be nice to be able to provide more context here, like the name of the file or information about the page/form where the action happened etc. I don't see how though 🤔

Reply:
We don't have more context inside this function. And we shouldn't make assumptions about it.

Can you please try that out in a testing environment with broken SVG?
You'll get pretty helpful hints - they're provided by that libxml message.
I gave some examples in an issue comment.